### ISSUE TITLE
TemporaryFolder can cause race condition due to initialization strategy

### ISSUE THREAD

**Issue by catchcase** [glytching/junit-extensions#13]
## Expected Behavior

Creation of the root directory should not unpredictably throw IO Exceptions when running tests which involve temporary folder operations in parallel.

## Actual Behavior

There may be a race condition due to how TemporaryFolder is initialized:

```
TemporaryFolder() {
    try {
      rootFolder = File.createTempFile(FILE_PREFIX, FILE_SUFFIX);
      Files.delete(rootFolder.toPath());
      Files.createDirectory(rootFolder.toPath());
    } catch (IOException ex) {
      throw new TemporaryFolderException("Failed to prepare root folder!", ex);
    }
}
```

This strategy of creating the file, deleting it, then calling createDirectory could seemingly cause IOExceptions. The issue occurs at random and passes when the test(s) is/are run another time.

Please see the following links for more information and why I think the above is suspect:

https://sefersezer.wordpress.com/2018/04/18/java-rules-file-createtempfile-should-not-be-used-to-create-a-directory/

https://rules.sonarsource.com/java/RSPEC-2976

## Specifications

  - JUnit-Extensions Version: 2.0.0 (Present in latest as well)

**Comment by glytching** [comment]
@catchcase: thanks for raising this. Can you provide any 'steps to reproduce'. It sounds like this is a sproadic issue so I suspect you won't have a 100% reliable reproduction case but do you have anything which could help direct the investigation?

**Comment by glytching** [comment]
I cannot reproduce the failure case but the change proposed in this issue's description here ...

* Is functionally equivalent: the test suite passes
* Is based on solid investigation elsewhere, see: https://rules.sonarsource.com/java/RSPEC-2976

So, I have applied the change and it will be released in v2.4.0
